### PR TITLE
refactor: Merge task and controller for create wallet flow

### PR DESCRIPTION
**Pull request by kartikCypherock**
Implements https://app.clickup.com/t/9002019994/PRF-4796

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4796 Task controller merge - New wallet](https://app.clickup.com/t/9002019994/PRF-4796)

File: src/menu/main_menu.c [hunk 1]
```diff
@@ -248,6 +248,8 @@ void main_menu_initialize(engine_ctx_t *ctx, const void *data_ptr) {
   return;
 }
 
+extern void create_wallet(void);
```
Comment (ujjwal-cyph) [inline src/menu/main_menu.c:251]: add a todo or put at a relevant place

File: src/level_three/add_wallet/controller/controller_new_wallet.c [hunk 2]
```diff
@@ -221,29 +296,36 @@ void generate_wallet_controller() {
       mnemonic_clear();
       memzero(wallet.wallet_share_with_mac_and_nonce,
               sizeof(wallet.wallet_share_with_mac_and_nonce));
-      flow_level.level_three = GENERATE_WALLET_SEED_GENERATED;
-    } break;
 
-    case GENERATE_WALLET_SEED_GENERATED: {
-      flow_level.level_three = GENERATE_WALLET_SAVE_WALLET_SHARE_TO_DEVICE;
-    } break;
+      instruction_scr_destructor();
```
Comment (amanCypherock) [inline src/level_three/add_wallet/controller/controller_new_wallet.c:300]: Remove `instruction_scr_destructor`, also on line 344

**Review by amanCypherock — CHANGES_REQUESTED** [review]
Please checkout the review comments here

File: src/level_three/add_wallet/controller/controller_new_wallet.c (hunk 2 above)
Comment (kartikCypherock) [reply src/level_three/add_wallet/controller/controller_new_wallet.c]: Fixed

File: src/level_three/add_wallet/controller/controller_new_wallet.c [hunk 3]
```diff
@@ -57,41 +57,74 @@
...
+ * @param current_state The current state of the flow which needs to be executed
+ * @return uint32_t The next state of the flow, based on processing the current
+ * state handler
+ */
+uint32_t create_wallet_state_handler(uint32_t current_state) {
+  uint32_t next_state = EXIT;
+
+  switch (current_state) {
+    case NAME_INPUT: {
+      input_text_init(
```
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply src/level_three/add_wallet/controller/controller_new_wallet.c:114]: Ok, makes sense, I have split them in 2 states.

File: src/level_three/add_wallet/controller/controller_new_wallet.c [hunk 4]
```diff
@@ -57,41 +57,74 @@
...
+#define SEED_GENERATE 8
+#define SEED_GENERATED 9
+#define SAVE_WALLET_SHARE_TO_DEVICE 10
+#define TAP_CARD_FLOW 11
+#define VERIFY_SHARES 12
+#define COMPLETED_WITHOUT_ERRORS 13
+#define COMPLETED_WITH_ERRORS 14
+#define TIMED_OUT 15
+#define EARLY_EXIT 16
+#define EXIT 17
```
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply src/level_three/add_wallet/controller/controller_new_wallet.c]: Yes, I have done it

File: common/core/ui_state_machine.h [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ * based on arguments
+ *
+ * @param state_on_text_input The state to be set on user text input
+ * @param state_on_rejection The state to be set on user rejection
+ * @param state_on_p0_event The state to be set on P0 event
+ * @return uint8_t The next state based on event received
+ */
+uint8_t text_input_next_state(uint8_t state_on_text_input,
+                              uint8_t state_on_rejection,
+                              uint8_t state_on_p0_event);
```
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply common/core/ui_state_machine.h]: Renamed as per suggestion, thanks

File: src/menu/main_menu.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply src/menu/main_menu.c]: Removed, will take this up seperately

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**